Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,472 @@
+
+            try
+            {
+                decimal previousBallance = balanceList[^2];
+                decimal currentBalance = balanceList[^1];
+                var loss = currentBalance - previousBallance;
+
+                return loss;
+            }
+            catch (IndexOutOfRangeException ex)
+            {
+                Debug.WriteLine(ex);
+            }
+            return default;
+        }
+
+        /// <summary>
+        /// Tries to convert a string to decimal.
+        /// </summary>
+        private static decimal ConvertStringToDecimal(string input)
+        {
+            decimal.TryParse(input, NumberStyles.Currency, numberFormatInfo, out decimal value);
+
+            return value;
+        }
+
+        /// <summary>
+        /// Converts the numbers split by comma part of the string to an array.
+        /// </summary>
+        private static decimal[] ConvertStringToDecimalArray(string personAndBalance)
+        {
+            if (!InputStringIsValid(personAndBalance))
+            {
+                return default;
+            }
+
+            var array = personAndBalance.Split(',');
+            var newArray = new decimal[array.Length - 1];
+
+            for (int i = 1; i < array.Length; i++)
+            {
+                newArray[i - 1] = ConvertStringToDecimal(array[i]);
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Finds the current balance of a single person in array.
+        /// </summary>
+        private static decimal CurrentBalanceForSinglePerson(string personAndBalance)
+        {
+            var balanceList = ConvertStringToDecimalArray(personAndBalance);
+            try
+            {
+                decimal currentBalance = balanceList[^1];
+
+                return currentBalance;
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
+            return default;
+        }
+
+        /// <summary>
+        /// Return index of Max value in decimal array.
+        /// </summary>
+        private static int FindDecimalArrayMax(decimal[] inputArray)
+        {
+            var max = decimal.Zero;
+            try
+            {
+                max = inputArray[0];
+            }
+            catch (IndexOutOfRangeException ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
+            var index = 0;
+            for (int i = index; i < inputArray.Length; i++)
+            {
+                if (inputArray[i] > max)
+                {
+                    max = inputArray[i];
+                    index = i;
+                }
+            }
+            return index;
+        }
+
+        /// <summary>
+        /// Return index of Min value in decimal array.
+        /// </summary>
+        private static int FindDecimalArrayMin(decimal[] inputArray)
+        {
+            var min = decimal.Zero;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'min'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4971928)